src/web/crate_details.rs

```diff
@@ -466,12 +469,194 @@ pub(crate) async fn get_all_releases(
     Ok(res.into_response())
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize)]
+struct ShortMetadata {
+    name: String,
+    version_or_latest: String,
+    doc_targets: Vec<String>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize)]
+struct PlatformList {
+    metadata: ShortMetadata,
+    inner_path: String,
+    use_direct_platform_links: bool,
+    current_target: String,
+}
+
+impl_axum_webpage! {
+    PlatformList = "rustdoc/platforms.html",
+    cpu_intensive_rendering = true,
+}
+
+#[tracing::instrument]
+pub(crate) async fn get_all_platforms_inner(
+    Path(params): Path<RustdocHtmlParams>,
+    Extension(pool): Extension<Pool>,
+    is_crate_root: bool,
+) -> AxumResult<AxumResponse> {
+    let req_path: String = params.path.unwrap_or_default();
+    let req_path: Vec<&str> = req_path.split('/').collect();
+
+    let release_found = match_version_axum(&pool, &params.name, Some(&params.version)).await?;
+    trace!(?release_found, "found release");
+
+    // Convenience function to allow for easy redirection
+    #[instrument]
+    fn redirect(
+        name: &str,
+        vers: &str,
+        path: &[&str],
+        cache_policy: CachePolicy,
+    ) -> AxumResult<AxumResponse> {
+        trace!("redirect");
+        // Format and parse the redirect url
+        Ok(axum_cached_redirect(
+            encode_url_path(&format!("/platforms/{}/{}/{}", name, vers, path.join("/"))),
+            cache_policy,
+        )?
+        .into_response())
+    }
+
+    let (version, version_or_latest) = match release_found.version {
+        MatchSemver::Exact((version, _)) => {
+            // Redirect when the requested crate name isn't correct
+            if let Some(name) = release_found.corrected_name {
+                return redirect(&name, &version, &req_path, CachePolicy::NoCaching);
+            }
+
+            (version.clone(), version)
+        }
+
+        MatchSemver::Latest((version, _)) => {
+            // Redirect when the requested crate name isn't correct
+            if let Some(name) = release_found.corrected_name {
+                return redirect(&name, "latest", &req_path, CachePolicy::NoCaching);
+            }
+
+            (version, "latest".to_string())
+        }
+
+        // Redirect when the requested version isn't correct
+        MatchSemver::Semver((v, _)) => {
+            // to prevent cloudfront caching the wrong artifacts on URLs with loose semver
+            // versions, redirect the browser to the returned version instead of loading it
+            // immediately
+            return redirect(&params.name, &v, &req_path, CachePolicy::ForeverInCdn);
```

Comment:
this redirect is cached, but the handlers are served below the `/-/` prefix, which is not part of the invalidation we run. 

So either we should use `NoCaching` here, or serve everything below the `/krate/` prefix, or invalidate additional paths.

Reply:
My alternative suggestion earlier might be good then, putting them with the rest of the non-rustdoc per-crate paths: `/crate/:name/:version/menus/...`